### DISCUSSION TITLE
GPU acceleration

### DISCUSSION THREAD

**Discussion by DanRRRR** [epochpic/epoch#500] [General]
Any plans to implement GPUs for code acceleration? That helps to get positive reviews for the access to high end supercomputers because they increase run efficiency by an order of magnitude with some PIC codes

Also some consumer videocards like NVIDIA 3090 and 4090 (and also the ones from  AMD ) recently became pretty capable and with decent amount of RAM. A lot of such hardware currently stays idle because of mining became not profitable. Are they usable for GPU acceleration?

**Comment by Status-Mirror** [comment]
Yes, but this is a long-term goal. Our main development task at the moment is a large-scale EPOCH rewrite to support libraries for running on exascale computers, which will likely involve a lot of GPU work.

**Comment by DanRRRR** [comment]
Hope on this route you will not chose the multi-language route. All the Fortran drivers/libraries for CUDA exists. Typically contaminating Fortran sources with any other languages lead to a dead end for any project and specifically for the top scientific codes where adopters are first scientists and engineers and not the professional programmers
